utils/testtools/crypto.go

```diff
@@ -135,72 +46,82 @@ func CreateCSR(key crypto.PrivateKey) (csr []byte, err error) {
 	}
 
-	defer os.RemoveAll(tmpDir)
+	publicKeyBytes, err := asn1.Marshal(*key.Public().(*rsa.PublicKey))
+	if err != nil {
+		return nil, nil, nil, aoserrors.Wrap(err)
+	}
 
-	caCertFile := path.Join(tmpDir, "ca.crt")
-	caKeyFile := path.Join(tmpDir, "ca.key")
-	csrFile := path.Join(tmpDir, "unit.csr")
-	csrConfFile := path.Join(tmpDir, "csr.conf")
-	unitCertFile := path.Join(tmpDir, "unit.der")
+	subjectKeyID := sha1.Sum(publicKeyBytes) // nolint:gosec
+	template.SubjectKeyId = subjectKeyID[:]
 
-	if err = ioutil.WriteFile(csrFile, csr, filePerm); err != nil {
-		return nil, aoserrors.Wrap(err)
+	if privateKey == nil {
+		privateKey = key
 	}
 
-	if err = ioutil.WriteFile(caCertFile, GetCACertificate(), filePerm); err != nil {
-		return nil, aoserrors.Wrap(err)
+	caBytes, err := x509.CreateCertificate(rand.Reader, template, parent, key.Public(), privateKey)
+	if err != nil {
+		return nil, nil, nil, aoserrors.Wrap(err)
 	}
 
-	if err = ioutil.WriteFile(caKeyFile, GetCAKey(), filePerm); err != nil {
-		return nil, aoserrors.Wrap(err)
+	cert, err := x509.ParseCertificate(caBytes)
+	if err != nil {
+		return nil, nil, nil, aoserrors.Wrap(err)
 	}
 
-	if err = ioutil.WriteFile(csrConfFile, []byte(csrConf), filePerm); err != nil {
-		return nil, aoserrors.Wrap(err)
-	}
+	certPEM := pem.EncodeToMemory(&pem.Block{
+		Type:  "CERTIFICATE",
+		Bytes: caBytes,
+	})
+
+	return cert, certPEM, key, nil
+}
 
-	var out []byte
+func Verify(leafCert *x509.Certificate, roots, intermediates []*x509.Certificate) error {
+	rootPool := x509.NewCertPool()
+	intermediatePool := x509.NewCertPool()
 
-	if out, err = exec.Command("openssl", "req", "-inform", "PEM", "-in",
-		csrFile, "-out", csrFile+".pem").CombinedOutput(); err != nil {
-		return nil, aoserrors.Errorf("message: %s, %s", string(out), err)
+	for _, c := range roots {
+		rootPool.AddCert(c)
 	}
 
-	if out, err = exec.Command("openssl", "x509", "-req", "-in", csrFile+".pem",
-		"-CA", caCertFile, "-CAkey", caKeyFile, "-extfile", csrConfFile, "-extensions", "ext",
-		"-outform", "PEM", "-out", unitCertFile, "-CAcreateserial", "-days", "3650").CombinedOutput(); err != nil {
-		return nil, aoserrors.Errorf("message: %s, %s", string(out), err)
+	for _, c := range intermediates {
+		intermediatePool.AddCert(c)
 	}
 
-	certData, err := ioutil.ReadFile(unitCertFile)
-	if err != nil {
-		return nil, aoserrors.Wrap(err)
+	opts := x509.VerifyOptions{
+		Roots:         rootPool,
+		Intermediates: intermediatePool,
 	}
 
-	caData, err := ioutil.ReadFile(caCertFile)
-	if err != nil {
-		return nil, aoserrors.Wrap(err)
+	if _, err := leafCert.Verify(opts); err != nil {
+		return aoserrors.Wrap(err)
 	}
 
-	certData = append(certData, caData...)
-	certData = append(certData, '\n')
+	return nil
+}
+
+func AppendCert(cert ...[]byte) (chain []byte) {
+	for _, c := range cert {
+		chain = append(chain, c...)
+	}
+
+	return chain
+}
+
+func RSAPrivateKeyToPEM(key *rsa.PrivateKey) []byte {
```

Comment:
There is crypturils.SavePrivateKeyToFile function. Is it not enough? If you really need convert private key to PEM format then:
1. implement generic function PrivateKeyToPEM in cryptutils (see crypturils.SavePrivateKeyToFile function as reference);
2. Use it inside crypturils.SavePrivateKeyToFile.